ns1/resource_datafeed.go

```diff
@@ -88,11 +95,44 @@ func DataFeedDelete(d *schema.ResourceData, meta interface{}) error {
 // DataFeedUpdate updates the given datafeed with modified parameters
 func DataFeedUpdate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
-	f := resourceDataToDataFeed(d)
+	f, err := resourceDataToDataFeed(d)
+	if err != nil {
+		return err
+	}
 	f.ID = d.Id()
 	if _, err := client.DataFeeds.Update(d.Get("source_id").(string), f); err != nil {
 		return err
 	}
 	dataFeedToResourceData(d, f)
 	return nil
 }
+
+//configAdapterIn adapts the configuration types
+func configAdapterIn(d *schema.ResourceData) error {
+	config := d.Get("config").(map[string]interface{})
+	if config != nil {
+		test_id := config["test_id"]
+		if test_id != nil {
+			test_id_int, err := strconv.Atoi(test_id.(string))
+			if err != nil {
+				return err
```

Comment:
I think we should wrap this error with a little more context about what failed, the default error is not going to be very helpful when troubleshooting a bad config in Terraform.